### PR TITLE
Fixes #5787

### LINKED ISSUE #5787: Search bar is broken
Selecting a element in the list when searching someone does nothing and throw an exception `TypeError: this.options is undefined`. It refers to this part of the code:

    else{ // The actual result
        this.options.element.val(formatted);
        window.location = data.url ? data.url : '/tags/' + data.name.substring(1);
    }

in the [search_view.js](https://github.com/diaspora/diaspora/blob/develop/app/assets/javascripts/app/views/search_view.js#L60) introduced #5761.

I guess this part would need some cukes.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/9611682-search-bar-is-broken?utm_campaign=plugin&utm_content=tracker%2F522&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F522&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by ghost**
Need help to write some tests on this.

### CODE REVIEW

**Comment by svbergerem** [comment]
I also started working on that issue and here are the tests I wrote so far:
https://gist.github.com/svbergerem/6cb1e93e9721932e68b1

Maybe that helps you writing some more.

**Comment by AugierLe42e** [comment]
Thanks.

**Comment by AugierLe42e** [comment]
I don't really understand how Cucumber works. Maybe someone could write an introduction to Cukes usage in d* on the wiki ?

**Comment by Flaburgan** [comment]
https://wiki.diasporafoundation.org/Testing_workflow is not enough? If so, it can be completed.

**Comment by AugierLe42e** [comment]
Testing_workflow is too long. I propose a new article on that question.
I think I've understood. You define the users stories in plain test in *features/{desktop|mobile}/*.feature* and you define the meaning of each sentence in *features/step_definitions/*.rb*, right ?

**Comment by jhass** [comment]
> Testing_workflow is too long.

No. Go read it.

**Comment by svbergerem** [comment]
@AugierLe42e For this PR I'd like to see two additional tests:
1. Search for a non-existent user. => you should be on the search page
2. Search for a hashtag

**Comment by AugierLe42e** [comment]
> No. Go read it.

Obviously, it depends on how one interprets *too long*.

@svbergerem : On it.

**Comment by goobertron** [comment]
Even I was able to understand the Testing workflow article...

**Comment by AugierLe42e** [comment]
I don't say it is not understandable. I say it is far from complete and useful...

**Comment by Flaburgan** [comment]
So, you're angry when there is no doc. When there is doc, you consider
it "far from useful" so, incomplete. But it's already too long? I'm
sorry but you look like trolling here. I know you aren't. But you know
the project for long enough now to know that shoot other people work is
useless and will not put the things forward.

**Comment by AugierLe42e** [comment]
I guess this topic is not clearly related to the PR and could be discussed elsewhere, am I wrong ?

**Comment by AugierLe42e** [comment]
Weird... [This test worked](https://travis-ci.org/diaspora/diaspora/jobs/54593921#L2166) when executed locally.

    When I click on the first search result
    Then I should be on the tag page for matrix

File: features/desktop/search.feature [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  Then I should see "Search for Trinity" within ".ac_even"
+
+  When I click on the first search result
+  Then I should see "Users matching TrinityLooking for posts tagged #Trinity?" within "#search_title"
+
+  When I enter "#Matrix" in the search input
+  Then I should see "#matrix" within ".ac_even"
+
+  When I click on the first search result
+  Then I should be on the tag page for matrix
```
Comment (svbergerem) [inline features/desktop/search.feature]: Try `Then I should be on the tag page for "matrix"`

File: features/desktop/search.feature [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  Then I should see "Alice Smith" within ".ac_results"
+
+  When I click on the first search result
+  Then I should see "Alice Smith" within ".profile_header #name"
+
+  When I enter "Trinity" in the search input
+  Then I should see "Search for Trinity" within ".ac_even"
+
+  When I click on the first search result
+  Then I should see "Users matching TrinityLooking for posts tagged #Trinity?" within "#search_title"
```
Comment (svbergerem) [inline features/desktop/search.feature]: `Then I should see "Users matching Trinity" within "#search_title"`
should be sufficient.

File: features/desktop/search.feature (hunk 1 above)
Comment (svbergerem) [inline features/desktop/search.feature]: Please separate these steps into three scenarios.

**Comment by AugierLe42e** [comment]
Looks good.
@svbergerem : Thanks for help.

**Comment by svbergerem** [comment]
@AugierLe42e https://github.com/diaspora/diaspora/pull/5788#discussion_r26515778

**Comment by AugierLe42e** [comment]
I apologize, I didn't see that one.

**Comment by svbergerem** [comment]
@AugierLe42e 
> And please change the commit message.

(I updated the comment)

**Comment by AugierLe42e** [comment]
> And please change the commit message.

To what ? Do you also want a rebase ?

**Comment by svbergerem** [comment]
Thank you!

**Pull request merged**